src/main/kotlin/no/nav/sykdig/digitalisering/papirsykmelding/NasjonalOppgaveService.kt

```diff
@@ -2,79 +2,263 @@ package no.nav.sykdig.digitalisering.papirsykmelding
-                        prognose = papirManuellOppgave.papirSmRegistering.prognose,
-                        utdypendeOpplysninger = papirManuellOppgave.papirSmRegistering.utdypendeOpplysninger,
-                        tiltakNAV = papirManuellOppgave.papirSmRegistering.tiltakNAV,
-                        tiltakArbeidsplassen = papirManuellOppgave.papirSmRegistering.tiltakArbeidsplassen,
-                        andreTiltak = papirManuellOppgave.papirSmRegistering.andreTiltak,
-                        meldingTilNAV = papirManuellOppgave.papirSmRegistering.meldingTilNAV,
-                        meldingTilArbeidsgiver = papirManuellOppgave.papirSmRegistering.meldingTilArbeidsgiver,
-                        kontaktMedPasient = papirManuellOppgave.papirSmRegistering.kontaktMedPasient,
-                        behandletTidspunkt = papirManuellOppgave.papirSmRegistering.behandletTidspunkt,
-                        behandler = papirManuellOppgave.papirSmRegistering.behandler,
-                    ),
-                utfall = null,
-                ferdigstiltAv = null,
-                datoFerdigstilt = null,
-                avvisningsgrunn = null,
-            )
+    fun oppdaterOppgave(sykmeldingId: String, utfall: String, ferdigstiltAv: String, avvisningsgrunn: String?): NasjonalManuellOppgaveDAO? {
+        val updated = nasjonalOppgaveRepository.findBySykmeldingId(sykmeldingId)?.copy(
+            utfall = utfall,
+            ferdigstiltAv = ferdigstiltAv,
+            avvisningsgrunn = avvisningsgrunn,
+            datoFerdigstilt = LocalDateTime.now(),
+            ferdigstilt = true,
+        )
+        when (updated) {
+            null -> log.info("Sykmelding $sykmeldingId not found ")
+            else -> nasjonalOppgaveRepository.save(updated)
+        }
+        return updated
+    }
 
-        if (existingId != null) {
-            nasjonalManuellOppgaveDAO.apply {
-                id = existingId
+    fun findByOppgaveId(oppgaveId: Int): NasjonalManuellOppgaveDAO? {
+        val oppgave = nasjonalOppgaveRepository.findByOppgaveId(oppgaveId)
+
+        if (oppgave == null) return null
+        return oppgave
+    }
+
+    fun getNasjonalOppgave(oppgaveId: String): NasjonalManuellOppgaveDAO {
+        val oppgave = findByOppgaveId(oppgaveId.toInt())
+        if (oppgave == null) {
+            log.warn("Fant ikke oppgave med id $oppgaveId")
+            throw NoOppgaveException("Fant ikke oppgave")
+        }
+        log.info("Hentet oppgave med id $oppgaveId")
+        return oppgave
+    }
+
+    suspend fun ferdigstillOppgave(
+        ferdigstillRegistrering: FerdigstillRegistrering,
+        beskrivelse: String?,
+        loggingMeta: LoggingMeta,
+        oppgaveId: String,
+    ) {
+        oppgaveClient.ferdigstillNasjonalOppgave(oppgaveId, ferdigstillRegistrering.sykmeldingId, ferdigstillRegistrering, loggingMeta)
+    }
+
+    fun avvisOppgave(
+        oppgaveId: Int,
+        request: String,
+        authorization: String,
```

Comment:
Kan fjerne `authorization`. I smregistrering blir `authorization` brukt for å sjekke om journalpost allerede er JOURNALFOERT eller FERDIGSTILT (+ logging). I syk-dig brukar vi berre journalpost for å sjekke det, i `ferdigstillNasjonalAvvistJournalpost`

Reply:
Meinar du at vi ikkje skal kreve autorisasjon på endepunkta?